From 761fc873a5089ecd9e95fc77511fa07338d05b44 Mon Sep 17 00:00:00 2001
From: Stef Walter <stefw@redhat.com>
Date: Fri, 10 Jan 2014 15:20:24 +0100
Subject: [PATCH] service: Don't crash if missing full_name_format in sssd.conf

Use the same default full_name_format as sssd.

We would like to retrieve this information from sssd, perhaps soon
we'll be able to do that via DBus.
---
 service/realm-sssd.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/service/realm-sssd.c b/service/realm-sssd.c
index 3fdd390..3b2678c 100644
--- a/service/realm-sssd.c
+++ b/service/realm-sssd.c
@@ -350,6 +350,10 @@ update_login_formats (RealmSssd *self)
 
 	/* Setup the login formats */
 	format = realm_ini_config_get (self->pv->config, self->pv->section, "full_name_format");
+	if (format == NULL)
+		format = realm_ini_config_get (self->pv->config, "sssd", "full_name_format");
+	if (format == NULL)
+		format = g_strdup ("%1$s@%2$s");
 
 	/* The full domain name */
 	domain_name = calc_domain (self);
-- 
1.8.4.2

